sdk/src/main/scala/io/horizen/storage/AbstractHistoryStorage.scala

```diff
@@ -145,11 +145,11 @@ abstract class AbstractHistoryStorage[
     var blockHasMcHeaders : Boolean = sidechainBlockInfo.mainchainHeaderBaseInfo.nonEmpty
 
     // go backwards on the chain until we found a mc header or we hit the genesis block, and break if we
-    // cross the threshold value of max revertable number of blocks
+    // cross the threshold value of max revertable number of blocks, included the latest block containing mc refs
     while(!blockHasMcHeaders) {
       scBlocksCount = scBlocksCount + 1
 
-      if (scBlocksCount >= params.maxHistoryRewritingLength ) {
+      if (scBlocksCount >= params.maxHistoryRewritingLength - 1 ) {
```

Comment:
Consider creating a constant like:
`val maxBlocksWithoutMcHeaders= params.maxHistoryRewritingLength - 1`
outside the while, so there is no need to calculated it at every loop. In addition, a comment could be added explaining why we can accept blocks without MC header for a max of maxHistoryRewritingLength - 1

Reply:
Done